src/snowflake/connector/cursor.py

```diff
@@ -704,6 +710,11 @@ def execute_async(self, *args, **kwargs):
         kwargs["_exec_async"] = True
         return self.execute(*args, **kwargs)
 
+    def describe(self, *args, **kwargs):
+        """Obtain the schema/description of the result without executing the query"""
+        kwargs["_describe_only"] = kwargs["_is_internal"] = True
```

Comment:
mark it as `is_internal` as well to be consistent with JDBC. https://github.com/snowflakedb/snowflake-jdbc/blob/69f560b9370cbe5de112f259600911ccd74333bf/src/main/java/net/snowflake/client/core/SFStatement.java#L135